src/os-freebsd.h

```diff
@@ -1,3 +1,11 @@
+#define _POSIX_C_SOURCE_TMP _POSIX_C_SOURCE
+#undef _POSIX_C_SOURCE
+#include <sys/param.h>
+#define _POSIX_C_SOURCE _POSIX_C_SOURCE_TMP
+#undef _POSIX_C_SOURCE_TMP
```

Comment:
Hello! This code is incorrect and dos not work.

You cannot define old value of `_POSIX_C_SOURCE` which was prior to `#undef _POSIX_C_SOURCE` line. C macros are evaluated at the time of usage. So above code after fifth line would result to having `_POSIX_C_SOURCE` defined to string `_POSIX_C_SOURCE_TMP`. It is because `_POSIX_C_SOURCE_TMP` is undefined after fifth line. You can check/verify it e.g. by calling C preprocessor only:

Content of `file.c`:
```c
#define _POSIX_C_SOURCE "OLD_STRING_VALUE"

#define _POSIX_C_SOURCE_TMP _POSIX_C_SOURCE
#undef _POSIX_C_SOURCE

#define _POSIX_C_SOURCE _POSIX_C_SOURCE_TMP
#undef _POSIX_C_SOURCE_TMP

int main() { printf("%s", _POSIX_C_SOURCE); }
```
And call:
```
$ cpp test.c 
# 1 "test"
# 1 "<built-in>"
# 1 "<command-line>"
# 31 "<command-line>"
# 1 "/usr/include/stdc-predef.h" 1 3 4
# 32 "<command-line>" 2
# 1 "test"
# 9 "test"
int main() { printf("%s", _POSIX_C_SOURCE_TMP); }
```
It is not `"OLD_STRING_VALUE"` anymore.